### PR TITLE
Update Spring Dependencies

### PR DESCRIPTION
**Pull request by moleske**
I'm mostly just curious if things still work after bumping (and it is hard to self deploy a pipeline)

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [x] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [x] Is your initial contribution a single, squashed commit?

- [x] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [x] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: boms/geode-all-bom/src/test/resources/expected-pom.xml [hunk 1]
```diff
@@ -670,91 +670,91 @@
       <dependency>
         <groupId>org.springframework.security</groupId>
         <artifactId>spring-security-config</artifactId>
-        <version>4.2.8.RELEASE</version>
+        <version>5.1.5.RELEASE</version>
```
Comment (kohlmu-pivotal) [inline boms/geode-all-bom/src/test/resources/expected-pom.xml]: Any reason why we not bump to `5.1.6`?

**Comment by kohlmu-pivotal** [comment]
Is there any reason why did not directly just tie the version to the `spring-framework-bom`?
`compile group: 'org.springframework', name: 'spring-framework-bom', version: '5.1.6.RELEASE', ext: 'pom'`

**Comment by moleske** [comment]
> Is there any reason why did not directly just tie the version to the `spring-framework-bom`?

nope, just lazy bumping numbers

**Comment by onichols-pivotal** [comment]
this has some overlap with https://github.com/apache/geode/pull/3390 so please review with @metatype

**Comment by moleske** [comment]
This bumped everything to Spring 5, so no real overlap. But i'm gonna close cause not actively looking (was just curious what was broken with spring 5)

**Pull request closed without merging**